modules/osdk-multi-arch-node-affinity.adoc

```diff
@@ -0,0 +1,121 @@
+// Module included in the following assemblies:
+//
+// * operators/operator_sdk/osdk-multi-arch-support.adoc
+
+:_content-type: PROCEDURE
+[id="osdk-multi-arch-node-affinity_{context}"]
+= Safe scheduling an Operator's node affinity for multiple architecture compute nodes
+
+Node affinity is a set of rules used by the scheduler to define a pod's placement. Set node affinity requirements to ensure that your Operator's workloads only schedule to compute nodes with architectures that are compatible with the images in the pod. 
+
+[WARNING]
+====
+If you do not set node affinity requirements and a container is scheduled to a compute node with an incompatible architecture, the pod fails with an `exec format` error. When the pod restarts and fails again with the same error, an `ImagePullBackoff` event occurs.
+====
+
+The following examples prevent the scheduling of pods to nodes with incompatible architectures. The `kubernetes.io/arch` and `kubernetes.io/os` keys reference the values of the architectures and operating systems defined in the manifest list.
+
+[IMPORTANT]
+====
+You must set node affinity in all objects that define a pod spec or pod template spec, including objects such as `Pod`, `Deployment`, `DaemonSet`, and `StatefulSet`.
+====
+
+.Prerequisites
+
+* An Operator project created or maintained with Operator SDK {osdk_ver_n1} or later
+
+.Procedure
+
+. Run the following command to search your Operator project for Kubernetes manifests that define `PodSpec` and `PodTemplateSpec` objects:
++
+[source,terminal]
+----
+$ grep <regex_for_PodSpec_and_PodTemplateSpec>
```

Comment:
Because object types in yaml don't actually declare their name, the only time you'd see PodSpec or PodTemplateSpec listed out explicitly like this would be when you interact with an API - like in the Go operator's dynamic workload example.

The dumb workaround I use is that I scan the code for the keyword `containers` since I know it is mandatory for this to be defined in both object types. I recognize this is not the cleanest strategy, but it's the best I've come up with.

Reply:
WDYT? Too hand-wavy?

https://github.com/openshift/openshift-docs/pull/64355/commits/a96f690fb2e136cab7956ac52c692a69eb964ba2

Reply:
Then maybe we highlight some YAML examples to help customers orient themselves?

Are there are any false positives for the `containers` field? If so, what happens if node affinity is set for an object that has the containers field but can't accept node affinity terms?